br/pkg/backup/client.go

```diff
@@ -533,14 +542,66 @@ func BuildBackupRangeAndSchema(
 			continue
 		}
 
-		tables, err := m.ListTables(dbInfo.ID)
+		tableNum := 0
+		err = m.IterTables(dbInfo.ID, func(tableInfo *model.TableInfo) error {
+			if !tableFilter.MatchTable(dbInfo.Name.O, tableInfo.Name.O) {
+				// Skip tables other than the given table.
+				return nil
+			}
+
+			schemasNum += 1
+			tableNum += 1
+			if buildRange {
+				tableRanges, err := BuildTableRanges(tableInfo)
+				if err != nil {
+					return errors.Trace(err)
+				}
+				for _, r := range tableRanges {
+					ranges = append(ranges, rtree.Range{
+						StartKey: r.StartKey,
+						EndKey:   r.EndKey,
+					})
+				}
+			}
+
+			return nil
+		})
+
 		if err != nil {
 			return nil, nil, nil, errors.Trace(err)
 		}
 
-		if len(tables) == 0 {
```

Comment:
Note: the origin code might have a little mistake: When `len(tables) > 0` but all the tables is skipped by `table-filter`, so the `dbInfo` won't be added into schemas